contrib/epee/src/net_ssl.cpp

```diff
@@ -240,11 +335,27 @@ boost::asio::ssl::context ssl_options_t::create_context() const
   {
     EVP_PKEY *pkey;
     X509 *cert;
-    CHECK_AND_ASSERT_THROW_MES(create_ssl_certificate(pkey, cert), "Failed to create certificate");
+    bool ok = false;
+
+    CHECK_AND_ASSERT_THROW_MES(create_ec_ssl_certificate(pkey, cert, NID_X9_62_prime256v1), "Failed to create certificate");
```

Comment:
Is this curve recommended somewhere? Or is it just the most widely supported?

Reply:
It is recommended in various places as good and the most compatible. It's also known as P-256 (and also secp256r1):

> crypto/ecdh/ecdhtest.c:        (NID_X9_62_prime256v1, "NIST Prime-Curve P-256", ctx, out))

https://security.stackexchange.com/questions/78621

My (Fedora) openssl has just a few curves available:
```
$ openssl ecparam -list_curves
  secp224r1 : NIST/SECG curve over a 224 bit prime field
  secp256k1 : SECG curve over a 256 bit prime field
  secp384r1 : NIST/SECG curve over a 384 bit prime field
  secp521r1 : NIST/SECG curve over a 521 bit prime field
  prime256v1: X9.62/SECG curve over a 256 bit prime field
```